06-documenting_R.Rmd

```diff
@@ -0,0 +1,110 @@
+
+## Guidelines for writing the system help files (in .Rd format)
+
+Following are the suggested guidelines while creating the system help files in the .Rd format. These are intended for writing the core documentations but may also be useful for package writers. (share documentation of a simple base function here?)
+
+There are three main parts of an .Rd file:
+
+1. **Header**: This part is for the basic information of the document/file. For instance, the name of the file, the topics documented, a title, a short textual description, and R usage information for the objects documented.
+
+2. **Body**: This part includes further information on the function's arguments and return value.
+
+3. **Footer**: This part is optional. Usually the keyword information is included here.
+
+All the above information is included in a .Rd file within a series of sections with standard nmes (user-defined sections are also allowed). These sections are discussed below:
+
+1. `\title` section:
+    * Capitalize each word.
+    * Do not end in a period.
+    * Avoid use of markup language (because markup language need not be suitable for various hypertext search systems).
+    
+2. `\usage` and `\examples` sections:
+    * Line length of 65 characters is advised.
+    * Use `TRUE` instead of `T` and `FALSE` instead of `F`.
+    * Add spaces around binary operators.
+    * Add spaces after commas in the argument lists.
+    * Use `<-` rather than `=` for assignments.
+    * Add spaces around the `<-` operator.
+    * Do not use tabs to indent (as these do not render correctly on all possible pagers).
+    * Use 4 spaces to indent the (example) code.
+    * Make sure the examples are directly executable.
+    * The examples should be system-independent.
+    * The examples should not require special facilities (for instance, Internet access or write permission to specific directories).
+    
+3. `\synopsis` section: The `\usage` section can have the function definition, or the actual definition can be included in the `\synopsis` section.
+
+4. `\source` and `\references` sections:
+    * Author(s) names should be written in the form `Author, A. B.`.
+    * Author(s) names should be separated by a comma or `and` (but not both).
+    * Separate paragraphs (separated by a blank line) should be used for each reference.
+    * Give a date immediatey after the author(s) names.
+    * Do not put a period after the date.
+    * Titles of books and journals (not articles) should be enclosed in `\emph{...}`.
+    * Volume numbers for journals are to be enclosed in `\bold{...}` and followed by a comma.
+    * Use `--` for page ranges.
+    * For given an address for a publisher use the format `New York: Springer-Verlag`.
+

```

Comment:
I think a paragraph about "See also" and families of functions could be interesting. As a way to make it easier for R users to navigate help pages. (Also I think there could be much improvement on this area).

Reply:
Yes, `\seealso` could be worth mentioning as a particular area where function documentation might be improved, since links to other functions may not have been considered when the documentation was originally written.

"family" is not an official part of the .Rd format, this is a feature of Roxygen documentation which adds a special sub-section to the "See also" section (https://cran.r-project.org/web/packages/roxygen2/vignettes/rd.html#cross-references). So I don't think this would be encouraged in contributions to the core codebase as it isn't consistent with the existing documentation (unless you have a counter-example @llrs?).

Reply:
I thought I had answered this... I didn't realize it was just a roxygen2 feature, then I agree it shouldn't be mentioned on this guide